fix: precedence of unary operators and expressions

MichaHoffmann opened the pull request:
Fixes #56 

We give index, splat and get_attr higher precedence then unary operations so that 

`!foo.bar`  reads as `!(foo.bar)` instead of `(!foo).bar`

MichaHoffmann commented:
@MXfive can you please sanity check?

MXfive commented:
> @MXfive can you please sanity check? - it seems to work on your provided testcases and seems reasonable generally, but just in case i missed something!

Awesome work!  I'll check this out tomorrow.

MXfive reviewed (approved):
LGTM in the codebase I discovered this issue on.

MichaHoffmann commented:
Thank you I'll bump the Terraform grammar then too and prepare a release in a bit

MichaHoffmann commented:
> @MichaHoffmann I just ran into this issue; is there something I can do to help get this merged?

Hey :wave:,

I rebased on latest main and added the fix to the  "terraform" grammar too. It should work, no idea whats wrong with the "Test parser" CI check though - that one comes from the org I think, @amaanq, do you have an idea?

MichaHoffmann commented:
We fail here: https://github.com/tree-sitter/node-tree-sitter/blob/3343ab8cf99540caf792eb69a29c243defffa983/index.js#L858 so i guess the node bindings are not compatible with code generated by the latest tree-sitter cli?

clason commented:
Looks like an issue with the `swift-tree-sitter` bindings. Try removing these lines: https://github.com/tree-sitter-grammars/tree-sitter-hcl/blob/009def4ae38ec30e5b40beeae26efe93484ab286/.github/workflows/ci.yaml#L41-L46

MichaHoffmann commented:
> Looks like an issue with the `swift-tree-sitter` bindings. Try removing these lines:
> 
> https://github.com/tree-sitter-grammars/tree-sitter-hcl/blob/009def4ae38ec30e5b40beeae26efe93484ab286/.github/workflows/ci.yaml#L41-L46
> 
> If that fails, just remove the Swift bindings from the test.

didnt help sadly! i removed the binding tests for swift from the actions
